File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.swift
class Solution {
    func reverseVowels(_ s: String) -> String {
        let vowels = Set(["a","e","i","o","u","A","E","I","O","U"])
        var stack = [String]()
        var sArr = s.map{String($0)}
        
        for i in sArr {
            if vowels.contains(i) {stack.append(i)}
        }

        for j in 0..<sArr.count {
            if vowels.contains(sArr[j]) {sArr[j] = stack.removeLast()}
        }

        return sArr.joined()
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = {'a','e','i','o','u', "A", 'E',"I","O","U"}
        record = []
        result = ""

        for char in s:
            if char in vowels:
                record.append(char)
        
        for i in range(len(s)):
            if s[i] in vowels:
                result += record.pop()
            else:
                result += s[i]
        
        return result